### PR TITLE
MCR-2213 RestAPIv2 simple search and filter endpoint (/explore)

### PR DESCRIPTION
**Pull request by rsteph-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2213).

### CODE REVIEW

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/model/MCRRestExploreResponseHeader.java  1
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/MCRLinkListJSONSerializer.java  2
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/MCRInstantXMLAdapter.java  1
- mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestExplore.java  18
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/model/MCRRestExploreResponseObject.java  2
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/model/MCRRestExploreResponse.java  1
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/model/MCRRestExploreResponsePayload.java  1
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/model/MCRRestExploreResponseLink.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5542460&bid=18170986)

File: mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestExplore.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                     sMaxAge = @MCRCacheControl.Age(time = 1, unit = TimeUnit.HOURS))
+    /*
+     * @Operation( summary = "Explore objects in this repository", responses
+     * = @ApiResponse( content = @Content(array = @ArraySchema(schema
+     * = @Schema(implementation = MCRObjectIDDate.class)))), tags =
+     * MCRRestUtils.TAG_MYCORE_OBJECT)
+     */
+    @XmlElementWrapper(
+            name = "mycoreobjects")
+    public Response exploreObjects(@QueryParam("start") String start, @QueryParam("rows") String rows,
```
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestExplore.java:89]: Codacy Issue found: [Avoid really long methods.](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5542460)

Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestExplore.java:89]: Codacy Issue found: [The method 'exploreObjects(String, String, String, List)' has an NPath complexity of 14400](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5542460)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on a06b13faad703bfa049ecd14abada49c316adb93! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5557-pull_request-972

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/model/MCRRestExploreResponseLink.java  1
- mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestExplore.java  8
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/model/MCRRestExploreResponsePayload.java  1
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/model/MCRRestExploreResponseHeader.java  1
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/MCRInstantXMLAdapter.java  1
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/model/MCRRestExploreResponseObject.java  2
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/model/MCRRestExploreResponse.java  1
- mycore-restapi/src/main/java/org/mycore/restapi/v2/explore/MCRLinkListJSONSerializer.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5542460&bid=18170986)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 1abaa04820c2fba95e71b783764cc2a05263a07e! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5565-pull_request-972

**Comment by rsteph-de** [comment]
We decided, that this endpoint should not be part of the MyCoRe core. The code will be moved to JSPDocportal and may be provided as plugin in the future. An other option would be to simulate this behaviour wih a customized Solr request handler.

**Pull request closed without merging**